src/com/example/helloworld/practice/recursion/ReverseDepthSum.java

```diff
@@ -0,0 +1,88 @@
+package com.example.helloworld.practice.recursion;
+
+import java.util.*;
+
+/**
+ * input: { {1,1}, 2, {1,1}} output: 8
+ * Given a nested list of integers, returns the sum of all integers in the list weighted by their REVERSED depth.
+ * * For example, given the list {{1,1},2,{1,1}} the deepest level is 2.
+ * Thus the function should return 8 (four 1's with weight 1, one 2 with weight 2)
+ * Given the list {1,{4,{6}}} the function should return 17 (one 1 with weight 3, one 4 with weight 2, and one 6 with weight 1)
+ */
+
+public class ReverseDepthSum {
+    public static class NestedInteger {
+        private Integer value;
+        private List<NestedInteger> list;
+
+        public NestedInteger(Integer val) {
+            value = val;
+        }
+
+        public NestedInteger(List<NestedInteger> nestedLists) {
+            list = nestedLists;
+        }
+
+        public boolean isInteger() {
+            return value != null;
+        }
+
+        public Integer getInteger() {
+            return value;
+        }
+
+        public List<NestedInteger> getList() {
+            return list;
+        }
+    }
+
+    public static int reverseDepthSum(List<NestedInteger> input) {
+        Map<Integer, Integer> levelMap = new TreeMap<Integer, Integer>(Collections.<Integer>reverseOrder());
```

Comment:
You can simplify this declaration as 

Map<Integer, Integer> levelMap = new TreeMap<>(Collections.reverseOrder());